docs/architecture/automated-testing.md

```diff
@@ -0,0 +1,20 @@
+# Automated Testing
+
+## Why is Puppeteer the tool of choice for end-to-end tests?
+
+There exists a rich ecosystem of tooling available for web-based end-to-end automated testing. Thus, it's a common question: "Why does Gutenberg use [Puppeteer](https://developers.google.com/web/tools/puppeteer/) instead of ([Cypress](https://cypress.io/), [Selenium](https://www.selenium.dev/), [Playwright](https://github.com/microsoft/playwright), etc)?". Given some historical unreliability of the build results associated with end-to-end tests, it's especially natural to weigh this question in considering whether our tools are providing more value than the effort required in maintaining them. While we should always be comfortable in reevaluating earlier decisions, there were and continue to be many reasons that Puppeteer is the best compromise of the options available for end-to-end testing.
+
+These include:
+
+- **Interoperability with existing testing framework**. Puppeteer is "just" a tool for controlling a Chrome browser, and makes no assumptions about how it's integrated into a testing environment. While this requires some additional effort in ensuring the test environment is available, it also allows for cohesion in how it integrates with an existing setup. Gutenberg is able to consistently use Jest for both unit testing and end-to-end testing. This is contrasted with other solutions like Cypress, which provide their own testing framework and assertion library as part of an all-in-one solution.
```

Comment:
I think now it supports Firefox by defaul too.

Reply:
Yes, it does 👍

Reply:
Yes. I wasn't sure how cohesive I could mention that here, without it overcomplicating the messaging. And the support is still pretty early. Puppeteer still markets itself as "Headless Chrome Node.js API", and "a Node library which provides a high-level API to control Chrome or Chromium over the DevTools Protocol. Puppeteer runs headless by default, but can be configured to run full (non-headless) Chrome or Chromium." The text largely reflects that.

Reply:
I guess I kinda alluded to it later in the document, "largely":

>The fact that Puppeteer largely targets the Chrome browser is non-ideal in how it does not provide full browser coverage.